### PR TITLE
More readable permission warning message

### LINKED ISSUE #1855: SNOW-1013234: Fix config permissions warning message
### What is the current behavior?

If permissions / owner of config file is incorrect, the command outputs long one-liner containing two commands (`chmod` and `chown`) to call to fix the issue.
1. Many users tend to overlook that there are two commands and give up after one does not fix the issue
2. In Snowflake CLI on MacOS default config path is inside "Application Support" directory - space inside the path turned out to be problematic for some users

```
/Users/pczajka/Snowflake/venv3.12/lib/python3.12/site-packages/snowflake/connector/config_manager.py:344: UserWarning: Bad owner or permissions on /Users/pczajka/Library/Application Support/snowflake/config.toml. To change owner, run `chown $USER /Users/pczajka/Library/Application Support/snowflake/config.toml`. To restrict permissions, run `chmod 0600 /Users/pczajka/Library/Application Support/snowflake/config.toml`.
  warn(f"Bad owner or permissions on {str(filep)}{chmod_message}")
  ```

### What is the desired behavior?

Reformattinmg the message to highlight that there are two commands to call (fix 1) and add quotes around filenames (fix 2).
For example:
```
/Users/pczajka/Snowflake/venv3.12/lib/python3.12/site-packages/snowflake/connector/config_manager.py:344: UserWarning: Bad owner or permissions on /Users/pczajka/Library/Application Support/snowflake/config.toml.
 * To change owner, run `chown $USER "/Users/pczajka/Library/Application Support/snowflake/config.toml"`.
 * To restrict permissions, run `chmod 0600 "/Users/pczajka/Library/Application Support/snowflake/config.toml"`.
  warn(f"Bad owner or permissions on {str(filep)}{chmod_message}")
  ```

### How would this improve `snowflake-connector-python`?

Improve user experience (also improve user experience of Snowflake CLI)

### References and other background

_No response_

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #1855

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Change format of bad config permissions warning message

### CODE REVIEW

**Comment by sfc-gh-pczajka** [comment]
I have read the CLA Document and I hereby sign the CLA

File: DESCRIPTION.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
   - Fixed PyArrow Table type hinting
   - Added support for connecting using an existing connection via the session and master token.
   - Added support for connecting to Snowflake by authenticating with multiple SAML IDP using external browser.
+  - Fixed bad config permissions warning message.
```
Comment (sfc-gh-sfan) [inline DESCRIPTION.md:20]: Let's instead say "Improved config permissions warning message"?

**Review by sfc-gh-sfan — APPROVED** [review]

**Review by sfc-gh-mkeller — APPROVED** [review]
Please add what @sfc-gh-sfan said, otherwise good to go!

:ship: :ship:

File: DESCRIPTION.md (hunk 1 above)
Comment (sfc-gh-pczajka) [reply DESCRIPTION.md]: Done :)

**Pull request merged**